tests/cache_file_generator.py

```diff
@@ -0,0 +1,6 @@
+import sys
+
+from test_cache_lock_file_perf import _acquire_lock_and_write_to_cache
```

Comment:
I would suggest to do it the other way around:
* Move `_acquire_lock_and_write_to_cache()` from that test runner to here
* Let the test runner import this helper function.

Not only because this would be more logically correct, it would also remove one unnecessary dependency - the `pytest` - when .Net team is going to use this generator as a stand-alone script.

Reply:
Hey Ray, you are right, this makes sense, I have made that change in my latest commit !